tests/templates/test_state_preparations/test_arbitrary_state_prep.py

```diff
@@ -0,0 +1,351 @@
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            qml.templates.ArbitraryStatePreparation(weights, wires=range(3))
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match="Weights tensor must be of shape"):
+            weights = np.zeros(12)
+            circuit(weights)
+
+
+class TestAttributes:
+    """Tests additional methods and attributes"""
+
+    @pytest.mark.parametrize(
+        "n_wires, expected_shape",
+        [
+            (3, (14,)),
+            (1, (2,)),
+            (2, (6,)),
+        ],
+    )
+    def test_shape(self, n_wires, expected_shape):
+        """Test that the shape method returns the correct shape of the weights tensor"""
+
+        shape = qml.templates.ArbitraryStatePreparation.shape(n_wires)
+        assert shape == expected_shape
+
+
+def circuit_template(weights):
+    qml.templates.ArbitraryStatePreparation(weights, range(2))
+    return qml.expval(qml.PauliZ(0))
+
+
+def circuit_decomposed(weights):
+    qml.PauliRot(weights[0], "XI", wires=[0, 1])
+    qml.PauliRot(weights[1], "YI", wires=[0, 1])
+    qml.PauliRot(weights[2], "IX", wires=[0, 1])
+    qml.PauliRot(weights[3], "IY", wires=[0, 1])
+    qml.PauliRot(weights[4], "XX", wires=[0, 1])
+    qml.PauliRot(weights[5], "XY", wires=[0, 1])
+    return qml.expval(qml.PauliZ(0))
+
+
+class TestInterfaces:
+    """Tests that the template is compatible with all interfaces, including the computation
+    of gradients."""
+
+    def test_list_and_tuples(self, tol):
+        """Tests common iterables as inputs."""
+
+        weights = [1] * 6
+
+        dev = qml.device("default.qubit", wires=2)
+
+        circuit = qml.QNode(circuit_template, dev)
+        circuit2 = qml.QNode(circuit_decomposed, dev)
+
+        res = circuit(weights)
+        res2 = circuit2(weights)
+
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+        weights_tuple = tuple(weights)
+        res = circuit(weights_tuple)
+        res2 = circuit2(weights_tuple)
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+    def test_autograd(self, tol):
+        """Tests the autograd interface."""
+
+        weights = np.random.random(size=(6,))
+        weights = pnp.array(weights, requires_grad=True)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        circuit = qml.QNode(circuit_template, dev)
+        circuit2 = qml.QNode(circuit_decomposed, dev)
+
+        res = circuit(weights)
+        res2 = circuit2(weights)
+        assert qml.math.allclose(res, res2, atol=tol, rtol=0)
+
+        grad_fn = qml.grad(circuit)
+        grads = grad_fn(weights)
+
+        grad_fn2 = qml.grad(circuit2)
+        grads2 = grad_fn2(weights)
+
+        assert np.allclose(grads[0], grads2[0], atol=tol, rtol=0)
+
+    def test_jax(self, tol, skip_if_no_jax_support):
+        """Tests the jax interface."""
+
+        import jax
```

Comment:
```suggestion
       jax = pytest.importorskip("jax")
```

We should probably condition the interface tests so people can tests out changes without the full configured environment.

Reply:
This is taken care of by the fixture `skip_if_no_jax_support`! I also find it strange but it works by just passing it to the test :) 

Do you think it is safer to do the "importorskip" in addition?

Reply:
Good point. I think I misunderstood the `skip_if_no_jax_support` fixture. But in the interest of being self-contained, it is better to just apply it directly!